src/pandas_profiling/model/summary_algorithms.py

```diff
@@ -36,11 +36,13 @@ def histogram_compute(
     stats = {}
     bins = config.plot.histogram.bins
     bins_arg = "auto" if bins == 0 else min(bins, n_unique)
-    stats[name] = np.histogram(finite_values, bins=bins_arg, weights=weights)
+    bins = np.histogram_bin_edges(finite_values, bins=bins_arg)
+    stats[name] = np.histogram(finite_values, bins=bins, weights=weights)
 
     max_bins = config.plot.histogram.max_bins
     if bins_arg == "auto" and len(stats[name][1]) > max_bins:
-        stats[name] = np.histogram(finite_values, bins=max_bins, weights=None)
+        bins_arg = np.histogram_bin_edges(finite_values, bins=max_bins)
+        stats[name] = np.histogram(finite_values, bins=bins_arg, weights=None)

```

Comment:
change the var name from bins_arg to bins to keep consistent

Reply:
Done